UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -954,15 +956,15 @@ public bool HasTile(Vector3Int position, LayerType layerType)
 
 			TileLocation tileLocation = null;
 			OverlayTile overlayTile = null;
-			position.z = 1;
+			position.z = 0;
```

Comment:
the reason it starts at one is because 0 is occupied by non-overlay tiles

Reply:
but since it's checking for a OverlayTile it doesn't technically clash

Reply:
Since it is incremented at the beginning of the while loop anyway the first position checked is when z = 1

Reply:
then the intention would be read better if you started at one and incremented the z at the end